### DISCUSSION TITLE
externally_connectable?

### DISCUSSION THREAD

**Discussion by machschnell** [mozilla/firefox-translations#378] [Q&A] [answered]
Can I make use of Firefox Translations from another extension via **externally_connectable**?

I've written an extension called **BBO Helper** (Firefox + Chromium) that improves the experience of playing on Bridge Base Online (BBO). BBO itself is well internationalized. However, there is a chat window used by players and kibitzers. I'd like to intercept chats that are not in the user's locale, translate them, and substitute in the translation, with the original included. So for example:

Wir haben Spiel im Herzen → We have game in hearts (Wir haben Wild im Herzen)

It would be great to do this locally, offering it as an option when Firefox Translation is also installed and the language conversion is supported.

**Comment by andrenatal** [comment, marked as answer]
Hi @machschnell, the easiest way to do that would be to incorporate the translation engine and the models the same way we did in the translations website [1] whose source is here [2]. Or you can look how we did here in the extension and try to replicate, but the translation website below is easier to understand.

[1] https://mozilla.github.io/translate/
[2]  https://github.com/mozilla/translate/
